### PR TITLE
changed stub endpoints to distinguish between GB and NI

### PR DESCRIPTION
**Pull request by RobertLaverick**

### CODE REVIEW

**Review by crystalmark — APPROVED** [review]

**Review by crystalmark — APPROVED** [review]

**Review by amaalali — CHANGES_REQUESTED** [review]
The context of this is to validate the bearer token headers. Is this a thing that should be done in this service? This is a stub service and it is usually advised that stubs have minimal logic

For the 2 different endpoints, I don't think we should have a separate controller methods since the original method had the functionality for both gb and ni (ignoring the above issue)

File: test/controllers/DeparturesControllerSpec.scala [hunk 1]
```diff
@@ -189,69 +189,109 @@ class DeparturesControllerSpec
...
+        val result = controller.nipost()(fakeUnauthorizedEmptyHeaderValueXmlRequest(routes.DeparturesController.nipost().url))
+        status(result) mustEqual UNAUTHORIZED
+      }
+    }
+
+    "POST IE015 with invalid bearer token" - {
+      "should return 401 Unauthorized" in {
+        val result = controller.nipost()(fakeUnauthorizedXmlRequest(routes.DeparturesController.nipost().url))
+        status(result) mustEqual UNAUTHORIZED
+      }
```
Comment (amaalali) [inline test/controllers/DeparturesControllerSpec.scala:292]: This is not the contract documented in the EIS spec

**Review by crystalmark — CHANGES_REQUESTED** [review]

File: test/controllers/DeparturesControllerSpec.scala [hunk 2]
```diff
@@ -189,69 +189,109 @@ class DeparturesControllerSpec
...
     }
-  }
 
-  "POST IE015 with invalid headers and valid bearer token" - {
-    "should return 202 Accepted" in {
-      val result = controller.post()(fakeInvalidXmlRequest)
-      status(result) mustEqual BAD_REQUEST
+    "POST IE015 with invalid headers and valid bearer token" - {
+      "should return 202 Accepted" in {
```
Comment (ashwinirudrappa) [inline test/controllers/DeparturesControllerSpec.scala:231]:
```suggestion
      "should return 400 BadRequest" in {
```
Please update the test description

File: test/controllers/DeparturesControllerSpec.scala [hunk 3]
```diff
@@ -189,69 +189,109 @@ class DeparturesControllerSpec
...
+  "nipost" - {
+    "POST IE015 with valid headers and valid bearer token" - {
+      "should return 202 Accepted" in {
+        val result = controller.nipost()(fakeValidXmlRequest(routes.DeparturesController.nipost().url, appConfig.eisniBearerToken))
+        status(result) mustEqual ACCEPTED
+      }
+    }
+
+    "POST IE015 with invalid headers and valid bearer token" - {
+      "should return 202 Accepted" in {
```
Comment (ashwinirudrappa) [inline test/controllers/DeparturesControllerSpec.scala:268]: @RobertLaverick Please update the test description to return 400 instead of 202

**Review by ashwinirudrappa — APPROVED** [review]
@RobertLaverick Code looks good from a web point of view

**Pull request merged**